src/plugins/manageQuickCss.tsx

```diff
@@ -0,0 +1,306 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+import { addContextMenuPatch, NavContextMenuPatchCallback, removeContextMenuPatch } from "@api/ContextMenu";
+import * as DataStore from "@api/DataStore";
+import { definePluginSettings, Settings } from "@api/Settings";
+import { CSSFileIcon } from "@components/Icons";
+import { Devs } from "@utils/constants";
+import definePlugin, { OptionType } from "@utils/types";
+import { Button, Menu, Toasts } from "@webpack/common";
+import React from "react";
+
+const enum AddStrategy {
+	Replace,
+	Append,
+	Prepend
+}
+
+const STORE_KEY = "quickCssSnippets";
+
+let cachedSnippetIds: string[] = [];
+
+const generateSnippetId = (messageSnowflake: string, snippet: string): string => {
+    const uniqueKey = `${messageSnowflake}-${snippet}`;
+    let hash = 0;
+
+    for (let i = 0; i < uniqueKey.length; i++) {
+        const char = uniqueKey.charCodeAt(i);
+        hash = ((hash << 5) - hash) + char;
+        hash |= 0;
+    }
+
+    return hash.toString();
+};
+
+
+const fetchSnippetIds = async () => {
+    cachedSnippetIds = await DataStore.get(STORE_KEY) || [];
+};
+
+const saveSnippetIds = async () => {
+    const storedSnippetIds = await DataStore.get(STORE_KEY) || [];
+    const newSnippetIds = cachedSnippetIds.filter(id => !storedSnippetIds.includes(id));
+    const updatedSnippetIds = [...storedSnippetIds, ...newSnippetIds];
+
+    await DataStore.set(STORE_KEY, updatedSnippetIds);
+};
+
+const addToSnippetIdCache = (snippetId: string) => {
+    if (!cachedSnippetIds.includes(snippetId)) {
+        cachedSnippetIds.push(snippetId);
+    }
+};
+
+const removeFromSnippetIdCache = (snippetId: string) => {
+    cachedSnippetIds = cachedSnippetIds.filter(id => id !== snippetId);
+};
+
+const importAllSnippets = async (snowflake: string, snippets: string[], strategy: AddStrategy) => {
+    for (const snippet of snippets) {
+        const snippetId = generateSnippetId(snowflake, snippet);
+        await importCssSnippet(snippetId, snippet, strategy);
+    }
+};
+
+const removeSnippets = async (snippetIds: string[]) => {
+    for (const snippetId of snippetIds) {
+        await removeCssSnippet(snippetId);
+    }
+};
+
+async function removeCssSnippet(snippetId: string) {
+    let quickCss = await VencordNative.quickCss.get();
+
+    const regex = new RegExp(
+        `\\/\\*\\nsnippet ${snippetId}[^]*?\\/\\* end snippet ${snippetId} \\*\\/`,
```

Comment:
Escaping / is pointless, no? And to avoid having to double escape, you can just use String.raw
```suggestion
        String.raw`/\*\nsnippet ${snippetId}[^]*?/\* end snippet ${snippetId} \*/`,
```